src/codeu/chat/client/commandline/Chat.java

```diff
@@ -136,8 +148,11 @@ public void invoke(Scanner args) {
     //
     panel.register("u-add", new Panel.Command() {
       @Override
-      public void invoke(Scanner args) {
-        final String name = args.hasNext() ? args.nextLine().trim() : "";
+      public void invoke(List<String> args) {
+        //TODO: possible place of error, not sure how to test username creation
+        // what's in the List being passed in? Is it already tokenized? The scanner code took the entire line?
+        // final String name = args.hasNext() ? args.nextLine().trim() : ""; (old/given code, uses a parameter Scanner args)
+        final String name = args.get(0) != null ? args.get(0) : ""; //this works if the args list passed in contains tokens
```

Comment:
`args.size() > 0` seems like a better check than `args.get(0) != null`. Technically, `get()` isn't guaranteed to return null: https://docs.oracle.com/javase/8/docs/api/java/util/List.html#get-int-

Same for the remaining instances.